examples/iam/rolepolicy.yaml

```diff
@@ -0,0 +1,23 @@
+---
+apiVersion: identity.aws.crossplane.io/v1alpha1
+kind: IAMRolePolicy
+metadata:
+  name: somerolepolicy
+spec:
+  forProvider:
+    PolicyName: external-name
+    RoleName: somerole
```

Comment:
`policyName` and `roleName`

Should `roleName` also support references?

Reply:
@chlunde - I had initially tried to use a reference resolver for the role.   The issue is that `IAMRole` is part of `identity/v1beta1` package and `IAMPolicy` is part of `identity/v1alpha1` package.
`identity/v1beta1` already depends on `identity/v1alpha1`   If a reference resolver (in `identity/v1alpha1`) is added for `IAMRole` (from `identity/v1beta1` it creates a circular dependency:
```
package github.com/crossplane/provider-aws/pkg/controller/identity/iamrolepolicy
	imports github.com/crossplane/provider-aws/apis/identity/v1alpha1
	imports github.com/crossplane/provider-aws/apis/identity/v1beta1
	imports github.com/crossplane/provider-aws/apis/identity/v1alpha1: import cycle not allowed
```
:(

Instead because `IAMRole` external-name is also the role name, I did not implement a reference resolver - it seemed redundant.

IMO all current resources which exist in `identity/v1alpha1` should be promoted to `identity/v1beta1` both for consistency reasons and also for the circular dependency issue above.

One other way this could be implemented is instead of `IAMRolePolicy` object, to simply allow policy definitions on the existing `IAMRole` resource, which would remove the need for reference resolver altogether.  I did not go that route, because this seemed like a better solution for a bunch of reasons.

Reply:
@chlunde - this does not appear to be an issue any longer now that everything has moved to `iam/v1beta1` - I have added the reference resolver back!